From 1d5ed58077c4a839d548c9951fd4626145d2da33 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 29 Oct 2019 15:00:26 +0000
Subject: [PATCH 0973/1077] MANTIS 0055465 CUSS 1.5.1 Biometrics Interface

added file CUSS-1.5.BIOMETRICS.GP.XSD
new version of CUSS-1.5.BIOMETRICS.XSD, CUSS.BIOMETRIC.XSD



git-svn-id: svn://localhost/SelfServiceCommon/trunk@1492 90b65887-3827-0410-9a23-83215b262276
---
 .../idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD       |  23 ++
 .../Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.XSD   |  76 ++--
 .../Massai/idl/Cuss/CUSS.BIOMETRIC.XSD        | 348 ++++++++++++++++--
 3 files changed, 391 insertions(+), 56 deletions(-)
 create mode 100644 SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD

diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
new file mode 100644
index 00000000..be12d972
--- /dev/null
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:bio="http://www.materna.com/CUSS/biometrics-1.0" targetNamespace="http://www.materna.com/CUSS/biometrics-1.0" elementFormDefault="qualified" attributeFormDefault="unqualified">
+	<xs:simpleType name="gp-parameter-name">
+		<xs:annotation>
+			<xs:documentation>
+				MATERNA specific keys for general purpose parameters values of CUSS-1.5.BIOMETRICS-20191007.xsd. 
+				It's not referenced by another schema but just a central 'documentation' of the parameters supported by the MATERNA CUSS platform.
+			</xs:documentation>
+		</xs:annotation>
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="materna.bcbp.raw">
+				<xs:annotation>
+					<xs:documentation><![CDATA[Raw barcode data string (e.g. 'M1SHORT/DARRYL         ES6QQR2LTNGVAEZY2051 142 006A0615 147>80B1MM7140BEZY2C                                         N  1684425666')]]></xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="materna.quality.frontalFace">
+				<xs:annotation>
+					<xs:documentation><![CDATA[Mimumim required FrontalFace quality for capturing facial images. Also used in the oposite direction for indicating the FrontalFace quality to the application]]></xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+		</xs:restriction>
+	</xs:simpleType>
+</xs:schema>
diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.XSD
index 4ef556b8..b0483b5b 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.XSD
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!-- Mit XMLSpy v2013 rel. 2 (http://www.altova.com) von Andreas Gehling (Materna GmbH) bearbeitet -->
-<xs:schema targetNamespace="http://www.iata.org/CUSS/biometrics-1.0" xmlns:bio="http://www.iata.org/CUSS/biometrics-1.0" xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" attributeFormDefault="unqualified">
+<!-- Mit XMLSpy v2020 (x64) (http://www.altova.com) von Materna Information & Communications SE (Materna Information & Communications SE) bearbeitet -->
+<xs:schema xmlns:bio="http://www.iata.org/CUSS/biometrics-1.0" xmlns:xs="http://www.w3.org/2001/XMLSchema" targetNamespace="http://www.iata.org/CUSS/biometrics-1.0" elementFormDefault="qualified" attributeFormDefault="unqualified">
 	<xs:simpleType name="biometric-provider-msg-type">
 		<xs:annotation>
 			<xs:documentation>Biometric provider message (document) type.</xs:documentation>
@@ -50,6 +50,27 @@ New increment number implements no functional changes (e.g. documentation change
 		</xs:annotation>
 		<xs:complexType>
 			<xs:choice>
+				<xs:element name="biometric-request-data">
+					<xs:annotation>
+						<xs:documentation>Supplementary device-acquired or other unsolicited data received and forwarded by application during biometric provider function execution.</xs:documentation>
+					</xs:annotation>
+					<xs:complexType>
+						<xs:choice>
+							<xs:element name="processing">
+								<xs:complexType>
+									<xs:choice>
+										<xs:element name="keypad-key" type="xs:string" maxOccurs="unbounded">
+											<xs:annotation>
+												<xs:documentation>A sequence of one or more keypad key identifiers as defined in the CUSS specification "Keyboard and Navigation Keypad" section. Values include, but are not limited to, NAVUP, NAVDOWN, NAVPREVIOUS, NAVNEXT, NAVENTER, NAVHOME, NAVEND and NAVHELP.</xs:documentation>
+											</xs:annotation>
+										</xs:element>
+									</xs:choice>
+								</xs:complexType>
+							</xs:element>
+							<xs:element ref="bio:passenger"/>
+						</xs:choice>
+					</xs:complexType>
+				</xs:element>
 				<xs:element name="biometric-request">
 					<xs:annotation>
 						<xs:documentation>Biometric function request from common use application to biometric provider component.</xs:documentation>
@@ -115,27 +136,6 @@ New increment number implements no functional changes (e.g. documentation change
 						</xs:sequence>
 					</xs:complexType>
 				</xs:element>
-				<xs:element name="biometric-request-data">
-					<xs:annotation>
-						<xs:documentation>Supplementary device-acquired or other unsolicited data received and forwarded by application during biometric provider function execution.</xs:documentation>
-					</xs:annotation>
-					<xs:complexType>
-						<xs:choice>
-							<xs:element name="processing">
-								<xs:complexType>
-									<xs:choice>
-										<xs:element name="keypad-key" type="xs:string" maxOccurs="unbounded">
-											<xs:annotation>
-												<xs:documentation>A sequence of one or more keypad key identifiers as defined in the CUSS specification "Keyboard and Navigation Keypad" section. Values include, but are not limited to, NAVUP, NAVDOWN, NAVPREVIOUS, NAVNEXT, NAVENTER, NAVHOME, NAVEND and NAVHELP.</xs:documentation>
-											</xs:annotation>
-										</xs:element>
-									</xs:choice>
-								</xs:complexType>
-							</xs:element>
-							<xs:element ref="bio:passenger"/>
-						</xs:choice>
-					</xs:complexType>
-				</xs:element>
 				<xs:element name="biometric-response">
 					<xs:annotation>
 						<xs:documentation>Function request response from biometric provider component to common use application.</xs:documentation>
@@ -168,9 +168,16 @@ New increment number implements no functional changes (e.g. documentation change
 				</xs:annotation>
 			</xs:attribute>
 			<xs:attribute name="biometric-function-type" use="required">
+				<xs:annotation>
+					<xs:documentation>What type of function to be executed (associate, enrol, identify, verify, ...) Pls. check enumeration.</xs:documentation>
+				</xs:annotation>
 				<xs:simpleType>
 					<xs:restriction base="xs:string">
+						<xs:enumeration value="snapshot"/>
+						<xs:enumeration value="associate"/>
+						<xs:enumeration value="disassociate"/>
 						<xs:enumeration value="enrol"/>
+						<xs:enumeration value="delete"/>
 						<xs:enumeration value="identify"/>
 						<xs:enumeration value="verify"/>
 					</xs:restriction>
@@ -331,17 +338,32 @@ New increment number implements no functional changes (e.g. documentation change
 						<xs:documentation>Passenger's family name (last name or surname).</xs:documentation>
 					</xs:annotation>
 				</xs:element>
-				<xs:element name="passenger-given-names" type="xs:string" minOccurs="0">
+				<xs:element name="passenger-given-names" type="xs:string" minOccurs="0" maxOccurs="unbounded">
 					<xs:annotation>
-						<xs:documentation>Passenger's given name (christian or first names).</xs:documentation>
+						<xs:documentation>Passenger's given names (christian or first names).</xs:documentation>
 					</xs:annotation>
 				</xs:element>
-				<xs:element name="passenger-mrz-data" type="xs:string" minOccurs="0">
+				<xs:element name="passenger-mrz-data" minOccurs="0" maxOccurs="3">
 					<xs:annotation>
 						<xs:documentation>MRZ data from passenger's government-issued travel, visa or residency document.</xs:documentation>
 					</xs:annotation>
+					<xs:complexType>
+						<xs:simpleContent>
+							<xs:extension base="xs:string">
+								<xs:attribute name="mrz-track" use="required">
+									<xs:simpleType>
+										<xs:restriction base="xs:unsignedInt">
+											<xs:enumeration value="1"/>
+											<xs:enumeration value="2"/>
+											<xs:enumeration value="3"/>
+										</xs:restriction>
+									</xs:simpleType>
+								</xs:attribute>
+							</xs:extension>
+						</xs:simpleContent>
+					</xs:complexType>
 				</xs:element>
-				<xs:element name="passenger-biometrics" minOccurs="0">
+				<xs:element name="passenger-biometrics" minOccurs="0" maxOccurs="unbounded">
 					<xs:annotation>
 						<xs:documentation>Passenger's biometrics image data.</xs:documentation>
 					</xs:annotation>
diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
index e2258780..d28300c6 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
@@ -1,5 +1,219 @@
 <xsd:schema xmlns:tns="urn:CUSS-1.5/types/biometrics" xmlns:xsd="http://www.w3.org/2001/XMLSchema" targetNamespace="urn:CUSS-1.5/types/biometrics" elementFormDefault="qualified" attributeFormDefault="unqualified">
     <!-- Global types -->
+    <xsd:simpleType name="genderType">
+        <xsd:annotation>
+            <xsd:documentation>Gender type codes.</xsd:documentation>
+        </xsd:annotation>
+        <xsd:restriction base="xsd:string">
+            <xsd:enumeration value="MALE"/>
+            <xsd:enumeration value="FEMALE"/>
+            <xsd:enumeration value="CHILD"/>
+            <xsd:enumeration value="UNKNOWN"/>
+        </xsd:restriction>
+    </xsd:simpleType>
+
+    <xsd:complexType name="nameDataType">
+        <xsd:annotation>
+            <xsd:documentation>A raw or converted name. The raw name can be in elements name or fName and lName. The converted name is stored in the parts list, title, etc.</xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:annotation>
+                <xsd:documentation>name data structure</xsd:documentation>
+            </xsd:annotation>
+            <xsd:element name="name" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>full name</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="fName" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>first name.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="fNameParts" type="xsd:string" minOccurs="0" maxOccurs="unbounded">
+                <xsd:annotation>
+                    <xsd:documentation>first name in form of split up and normalized (“ä”-> “ae”, etc.) name parts.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="lName" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>last name.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="lNameParts" type="xsd:string" minOccurs="0" maxOccurs="unbounded">
+                <xsd:annotation>
+                    <xsd:documentation>last name in form of split up and normalized (“ä”-> “ae”, etc.) name parts.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="gender" type="tns:genderType" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>gender</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+
+    <xsd:complexType name="passportData">
+        <xsd:annotation>
+            <xsd:documentation>Passport document data structure</xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element name="name" type="tns:nameDataType" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Name data structure</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="documentNumber" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Document number</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="expiryDate" type="xsd:date" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Expiry date of this document</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="countryOfIssue" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>ISO code of document issuing country</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="cityOfIssue" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>nameof issuing city</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="dateOfBirth" type="xsd:date" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Date of birth</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="nationality" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Document owner's nationality</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="dateOfIssue" type="xsd:date" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Issue date of this document</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="mrzTrack" type="xsd:string" minOccurs="0" maxOccurs="unbounded">
+                <xsd:annotation>
+                    <xsd:documentation>List of "machine readable zone" tracks</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="passportImage" type="xsd:base64Binary" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Base64 encoded passport image data</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+
+    <xsd:complexType name="boardingPassData">
+        <xsd:annotation>
+            <xsd:documentation>Passport document data structure</xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element name="name" type="tns:nameDataType" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Name data structure</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <!-- flight related data -->
+            <xsd:element name="airlineCode" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>2L-Airline code</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="flightNumber" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Flight number in trimmed format, may include flight number suffix</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="flightDate" type="xsd:date" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Full flight date</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="origin" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Origin 3L-code</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="destination" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Destination 3L-code</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <!-- pax related data -->
+            <xsd:element name="sequenceNumber" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Boarding number or standby number for this segment, if any. The type is determined by the checkinStatus.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="pnr" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>PNR code (file key) for this segment.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="compartmentCode" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Passenger's compartment code (e.g. 'F')</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="seat" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Passenger's seat (e.g. '015A)</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="barcodeString" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>List of "machine readable zone" tracks</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+    
+    <xsd:complexType name="pax">
+        <xsd:annotation>
+            <xsd:documentation></xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element name="id" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Passenger id, if applicable</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="hostNativeId" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Host specific passenger id, if applicable</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="name" type="tns:nameDataType" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Name data structure</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="passport" type="tns:passportData" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>TBC</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="boardingPass" type="tns:boardingPassData" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>TBC</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="facialImage" type="xsd:base64Binary" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Base64 encoded facial image data. Required in case of identification via facial image</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+    
     <xsd:simpleType name="score">
         <xsd:annotation>
             <xsd:documentation>Comparision or match score</xsd:documentation>
@@ -17,51 +231,66 @@
             </xsd:maxInclusive>
         </xsd:restriction>
     </xsd:simpleType>
-    
+
     <!-- Command element types -->
     <xsd:simpleType name="commandType">
         <xsd:annotation>
             <xsd:documentation>Possible camera commands</xsd:documentation>
         </xsd:annotation>
         <xsd:restriction base="xsd:string">
-            <xsd:enumeration value="getBiometricImage" >
+            <xsd:enumeration value="getBiometricImage">
                 <xsd:annotation>
                     <xsd:documentation>Get biometric face image. Returns a quality rated face image with the minimum given absolute quality value (threshold)</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
-            <xsd:enumeration value="getIdentifiedBiometricImage" >
+            <xsd:enumeration value="getIdentifiedBiometricImage">
                 <xsd:annotation>
                     <xsd:documentation>Get biometric face image which was identified by the matching server. This command only returns data in case of a previous enrollment operation</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
-            <xsd:enumeration value="getActualImage" >
+            <xsd:enumeration value="getActualImage">
                 <xsd:annotation>
                     <xsd:documentation>Get biometric face image. Returns a *non* quality rated face image. Nothing will be returned, if no face is detected</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
-            <xsd:enumeration value="compareImages" >
+            <xsd:enumeration value="compareImages">
                 <xsd:annotation>
                     <xsd:documentation>Compares two given images (e.g. for ID check). Requires comparison data in element compareData</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
-            <xsd:enumeration value="compareImageWithCameraImage" >
+            <xsd:enumeration value="compareImageWithCameraImage">
                 <xsd:annotation>
                     <xsd:documentation>Compares given images against camera image. Requires comparison data in element compareData</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
-            <xsd:enumeration value="enrollPerson" >
+            <xsd:enumeration value="enrollPerson">
                 <xsd:annotation>
                     <xsd:documentation>Perform biometric enrollment with given data. Requires enrollment data in element enrollmentData</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
-            <xsd:enumeration value="deleteEnrollment" >
+            <xsd:enumeration value="deleteEnrollment">
                 <xsd:annotation>
                     <xsd:documentation>Deltes enrollment and template data for an already enrolled person</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
+            <xsd:enumeration value="publishPresence">
+                <xsd:annotation>
+                    <xsd:documentation>Publish presence of passenger at a certain touch-point (e.g. e-gate or SBD unit)</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="error">
+                <xsd:annotation>
+                    <xsd:documentation>Error occured (only applicable for response commands)</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="identify">
+                <xsd:annotation>
+                    <xsd:documentation>Identifies a passenger. What passenger details are retured depends on the implementation</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
         </xsd:restriction>
     </xsd:simpleType>
-    
+
     <xsd:simpleType name="qualityMetricType">
         <xsd:annotation>
             <xsd:documentation>The quality metrics types of an image</xsd:documentation>
@@ -164,7 +393,7 @@
             </xsd:enumeration>
         </xsd:restriction>
     </xsd:simpleType>
-    
+
     <xsd:simpleType name="qualityMetricValue">
         <xsd:restriction base="xsd:integer">
             <xsd:minInclusive value="-1">
@@ -221,14 +450,14 @@
             <xsd:documentation>Data element for command enrollPerson</xsd:documentation>
         </xsd:annotation>
         <xsd:sequence>
-            <xsd:element name="identCode" type="xsd:base64Binary">
+            <xsd:element name="identCode" type="xsd:base64Binary" minOccurs="0">
                 <xsd:annotation>
                     <xsd:documentation>Base64 encoded identification data (e.g. barcode)</xsd:documentation>
                 </xsd:annotation>
             </xsd:element>
             <xsd:element name="image" type="xsd:base64Binary">
                 <xsd:annotation>
-                    <xsd:documentation>Base64 encoded image data</xsd:documentation>
+                    <xsd:documentation>Base64 encoded image data. In case of multiple image data (e.g. facial-image and passport-image), this should be the facial-image data</xsd:documentation>
                 </xsd:annotation>
             </xsd:element>
         </xsd:sequence>
@@ -256,7 +485,25 @@
             </xsd:element>
         </xsd:sequence>
     </xsd:complexType>
-    
+
+    <xsd:complexType name="publishPresenceData">
+        <xsd:annotation>
+            <xsd:documentation>Data element for command publishPresence</xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element name="position" type="xsd:string">
+                <xsd:annotation>
+                    <xsd:documentation>Name / id of position passenger's presence should be published for</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="timestamp" type="xsd:dateTime" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>timestamp of presence event</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+
     <xsd:complexType name="command">
         <xsd:annotation>
             <xsd:documentation></xsd:documentation>
@@ -264,7 +511,7 @@
         <xsd:sequence>
             <xsd:element name="id" type="xsd:string">
                 <xsd:annotation>
-                    <xsd:documentation>Command id reference</xsd:documentation>
+                    <xsd:documentation>Command id reference. Helpful if referencing session and screen id included (e.g. 'AR6G3fY6yFpQuH13cq1OX5-identify.kiosk.enterIdData-1')</xsd:documentation>
                 </xsd:annotation>
             </xsd:element>
             <xsd:element name="type" type="tns:commandType">
@@ -302,22 +549,60 @@
                     <xsd:documentation>Required data in case of revoking an enrollment</xsd:documentation>
                 </xsd:annotation>
             </xsd:element>
+            <xsd:element name="publishPresenceData" type="tns:publishPresenceData" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Required data in case of publishing passenger presence</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
         </xsd:sequence>
     </xsd:complexType>
 
     <!-- Command response element types -->
     <xsd:simpleType name="commandResult">
         <xsd:restriction base="xsd:string">
-            <xsd:enumeration value="success"/>
-            <xsd:enumeration value="timeout"/>
-            <xsd:enumeration value="cameraError"/>
-            <xsd:enumeration value="invalidConfiguration"/>
-            <xsd:enumeration value="matchSuccess"/>
-            <xsd:enumeration value="matchFailed"/>
-            <xsd:enumeration value="failed"/>
+            <xsd:enumeration value="success">
+                <xsd:annotation>
+                    <xsd:documentation>Executing the requested command was succesful</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="timeout">
+                <xsd:annotation>
+                    <xsd:documentation>Executing the requested command timed out</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="cameraError">
+                <xsd:annotation>
+                    <xsd:documentation>Camera error occured</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="invalidConfiguration">
+                <xsd:annotation>
+                    <xsd:documentation>Configuration is invalid</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="matchSuccess">
+                <xsd:annotation>
+                    <xsd:documentation>Comparing images was successful. See score for matching score</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="matchFailed">
+                <xsd:annotation>
+                    <xsd:documentation>Images do not match and requested minimum score was not reached</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="failed">
+                <xsd:annotation>
+                    <xsd:documentation>Operation failed for some reason. See message for further details</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="notIdentified">
+                <xsd:annotation>
+                    <xsd:documentation>Passenger is unknown and cannt be identifies (i.e. not enrolled before)</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
         </xsd:restriction>
     </xsd:simpleType>
-    
+
     <xsd:complexType name="commandResponse">
         <xsd:annotation>
             <xsd:documentation>command response data type</xsd:documentation>
@@ -340,7 +625,7 @@
             </xsd:element>
             <xsd:element name="message" type="xsd:string" minOccurs="0">
                 <xsd:annotation>
-                    <xsd:documentation>Error message if available</xsd:documentation>
+                    <xsd:documentation>Internal error message or error description from biometrics backend if available</xsd:documentation>
                 </xsd:annotation>
             </xsd:element>
             <xsd:element name="score" type="tns:score" minOccurs="0">
@@ -378,14 +663,14 @@
             <xsd:enumeration value="both eyes"/>
         </xsd:restriction>
     </xsd:simpleType>
-    
+
     <xsd:simpleType name="imageFormat">
         <xsd:restriction base="xsd:string">
             <xsd:enumeration value="png"/>
             <xsd:enumeration value="jpg"/>
         </xsd:restriction>
     </xsd:simpleType>
-    
+
     <xsd:simpleType name="imageType">
         <xsd:restriction base="xsd:string">
             <xsd:enumeration value="thumbnail"/>
@@ -394,7 +679,7 @@
             <xsd:enumeration value="photo"/>
         </xsd:restriction>
     </xsd:simpleType>
-    
+
     <xsd:simpleType name="captureResult">
         <xsd:restriction base="xsd:string">
             <xsd:enumeration value="success"/>
@@ -404,7 +689,7 @@
             <xsd:enumeration value="tooBad"/>
         </xsd:restriction>
     </xsd:simpleType>
-    
+
     <xsd:complexType name="capture">
         <xsd:annotation>
             <xsd:documentation>capture data main element</xsd:documentation>
@@ -467,7 +752,7 @@
             </xsd:element>
         </xsd:sequence>
     </xsd:complexType>
-    
+
     <xsd:complexType name="image">
         <xsd:annotation>
             <xsd:documentation>image</xsd:documentation>
@@ -510,13 +795,18 @@
             </xsd:element>
         </xsd:sequence>
     </xsd:complexType>
-    
+
     <!-- Global element sub-types -->
     <xsd:complexType name="biometrics">
         <xsd:annotation>
             <xsd:documentation>biometric root type</xsd:documentation>
         </xsd:annotation>
         <xsd:sequence>
+            <xsd:element name="pax" type="tns:pax" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>passenger data, if applicable</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
             <xsd:element name="command" type="tns:command" minOccurs="0">
                 <xsd:annotation>
                     <xsd:documentation>list of commands</xsd:documentation>
-- 
2.41.0.windows.1

